### DISCUSSION TITLE
What is the best way to trigger an emergency landing ?

### DISCUSSION THREAD

**Discussion by MartinSegaert** [ctu-mrs/mrs_uav_system#161] [Q&A]
Hello,

I am planning to do hardware tests with 2 UAVs transporting a bar payload. As they are physically linked, it is quite dangerous. Therefore I am implementing a safety feature to ensure that both UAVs do an emergency landing if a problem occurs or if one of them is already doing an emergency landing.

For now I can:
- Manually command an emergency landing of each UAV individually from the terminal with the "eland" service
- Manually command a simultaneous emergency landing of both UAVs from the terminal with the "eland" service with the nimbro network
- Automatically trigger an emergency landing (for each UAV individually) from our own controller or tracker by returning an empty command. This makes the control manager trigger an emergency landing. Thanks to communication between the two UAVs, I am able ,with this method, to make both UAVs do an emergency landing whenever I want.

However, my question is:
Is there a better or cleaner way to trigger an emergency landing from our controller or tracker than by returning an empty command ? Can the eland service be used in a our controller or tracker code instead of in the terminal. And if yes, can the eland service be used to trigger the eland of both UAVs with nimbro, like it is done in the terminal ?

Thanks again for your help,

Martin

**Comment by klaxalk** [comment]
Hey, I would **not** recommend calling the service from within the controller's/tracker's update() method. The method is under a lot of mutexes within the control manager and it is possible that you could cause a deadlock doing this. I would either suggest returning the empty command, that is completely valid. Alternatively, since you need to synchronize it among the two drones, I would probably publish some status messages from the controller. The status could be subscribed by a node handling the synchronization between the drones. This node would call the land as a result of either of the two controller's signaling eland...
